recipes/solr-deploy.rb

```diff
@@ -40,23 +40,40 @@
 end
 
 app = search("aws_opsworks_app", "shortname:#{node['datashades']['app_id']}-#{node['datashades']['version']}-solr*").first
-download_url = app['app_source']['url']
-solr_version = download_url[/\/solr-([^\/]+)[.]zip$/, 1]
-
-unless (::File.directory?("/opt/solr-#{solr_version}") and ::File.symlink?("/opt/solr") and ::File.readlink("/opt/solr").eql? "/opt/solr-#{solr_version}")
-	remote_file "#{Chef::Config[:file_cache_path]}/solr.zip" do
-		source app['app_source']['url']
-	end
-
-	bash "install #{service_name} #{solr_version}" do
-		user "root"
-		code <<-EOS
-		unzip -u -q #{Chef::Config[:file_cache_path]}/solr.zip -d /tmp/solr
-		mv #{Chef::Config[:file_cache_path]}/solr.zip #{Chef::Config[:file_cache_path]}/solr-#{solr_version}.zip
-		cd /tmp/solr/solr-#{solr_version}
-		/tmp/solr/solr-#{solr_version}/bin/install_solr_service.sh #{Chef::Config[:file_cache_path]}/solr-#{solr_version}.zip -f
-		EOS
-	end
+solr_version = app['app_source']['url'][/\/solr-([^\/]+)[.]zip$/, 1]
+installed_solr_version = "/opt/solr-#{solr_version}"
+
+unless ::File.identical?(installed_solr_version, "/opt/solr")
```

Comment:
looks good but won't fix prod. are we rotating solr servers with our next release so it does work?

Reply:
Actually, I had a plan for that. There is a new 7.7.3 patch release out; we can update to that and trigger a fresh install.